src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/events/EndOfServiceReportEventPublisher.kt

```diff
@@ -1,38 +1,25 @@
 package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events
 
-import org.springframework.context.ApplicationEvent
 import org.springframework.context.ApplicationEventPublisher
 import org.springframework.stereotype.Component
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.LocationMapper
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller.EndOfServiceReportController
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.EndOfServiceReport
 
-enum class EndOfServiceReportEventType {
-  SUBMITTED,
-}
-
-class EndOfServiceReportEvent(
-  source: Any,
-  val type: EndOfServiceReportEventType,
-  val endOfServiceReport: EndOfServiceReport,
-  val detailUrl: String
-) :
-  ApplicationEvent(source) {
-  override fun toString(): String {
-    return "EndOfServiceReportEvent(type=$type, referralId=${endOfServiceReport.referral.id}, detailUrl='$detailUrl', source=$source)"
-  }
-}
-
 @Component
 class EndOfServiceReportEventPublisher(
   private val applicationEventPublisher: ApplicationEventPublisher,
   private val locationMapper: LocationMapper
 ) {
-  fun endOfServiceReportSubmittedEvent(endOfServiceReport: EndOfServiceReport) {
+  fun publishSubmittedEvent(endOfServiceReport: EndOfServiceReport) {
     applicationEventPublisher.publishEvent(
-      EndOfServiceReportEvent(
-        this, EndOfServiceReportEventType.SUBMITTED,
-        endOfServiceReport, getEndOfServiceReportUrl(endOfServiceReport)
+      EndOfServiceReportSubmittedEvent(
+        this,
+        endOfServiceReport.referral.id,
+        getEndOfServiceReportUrl(endOfServiceReport),
+        endOfServiceReport.id,
+        endOfServiceReport.submittedAt!!,
+        endOfServiceReport.submittedBy!!
```

Comment:
Main change: switch from passing down `EndOfServiceReport` to the event and pass down simpler data points: IDs, dates, and `AuthUser`. This forces the publisher to ensure we broadcast an event representing what happened.